File: app/src/main/java/com/skyline/app/network/Blog.java
package com.skyline.app.network;

import java.io.Serializable;
import java.util.List;

public class Blog implements Serializable {
    public String blogCode;
    public String title;
    public String slug;
    public String category;
    public String categorySlug;
    public String destination;
    public String destinationCode;
    public String shortDescription;
    public String introContent;
    public String thumbnailUrl;
    public String coverImageUrl;
    public Author author;
    public String publishedDate;
    public String readTime;
    public int likesCount;
    public int viewsCount;
    public List<QuickInfo> quickInfos;
    public List<Section> sections;
    public CTA cta;
    public boolean isFeatured;

    public static class Author implements Serializable {
        public String authorId;
        public String name;
        public String avatarUrl;
    }

    public static class QuickInfo implements Serializable {
        public String quickInfoId;
        public String icon;
        public String title;
        public String value;
    }

    public static class Section implements Serializable {
        public String sectionId;
        public int sectionNumber;
        public String title;
        public String type;
        public List<SectionItem> items;
    }

    public static class SectionItem implements Serializable {
        public String itemId;
        public String title;
        public String subtitle;
        public String description;
        public String imageUrl;
        public String icon;
        public List<String> bulletPoints;
    }

    public static class CTA implements Serializable {
        public String text;
        public String action;
        public String destinationCode;
    }
}

File: backend/models/Blog.js
const mongoose = require("mongoose");

const QuickInfoSchema = new mongoose.Schema({
  quickInfoId: String,
  icon: String,
  title: String,
  value: String,
  sortOrder: Number
});

const SectionItemSchema = new mongoose.Schema({
  itemId: String,
  title: String,
  subtitle: String,
  description: String,
  imageUrl: String,
  icon: String,
  bulletPoints: [String],
  sortOrder: Number
});

const SectionSchema = new mongoose.Schema({
  sectionId: String,
  sectionNumber: Number,
  title: String,
  type: String, // reason_grid, place_grid, food_grid, tips, itinerary
  sortOrder: Number,
  items: [SectionItemSchema]
});

const BlogSchema = new mongoose.Schema({
  blogCode: { type: String, required: true, unique: true },
  title: { type: String, required: true },
  slug: { type: String, required: true, unique: true },
  category: String,
  categorySlug: String,
  destination: String,
  destinationCode: String,
  shortDescription: String,
  introContent: String,
  thumbnailUrl: String,
  coverImageUrl: String,
  author: {
    authorId: String,
    name: String,
    avatarUrl: String
  },
  publishedDate: Date,
  readTime: String,
  likesCount: { type: Number, default: 0 },
  viewsCount: { type: Number, default: 0 },
  quickInfos: [QuickInfoSchema],
  sections: [SectionSchema],
  cta: {
    text: String,
    action: String,
    destinationCode: String
  },
  isFeatured: { type: Boolean, default: false },
  status: { type: String, enum: ["draft", "published"], default: "published" },
  sortOrder: Number
}, { timestamps: true });

module.exports = mongoose.model("Blog", BlogSchema);
